test/integration/test_download.py

```diff
@@ -95,6 +95,15 @@ def testDownloadNonExistingArtifact(self):
         with pytest.raises(AttributeError):
             self.download('-n', '--release', '5.3', '--ice', '3.3')
 
+    def testDownloadBuildNumber(self):
+        # Old Jenkins artifacts are deleted so we can't download.
+        # Instead assert that an AttributeError is raised.
+        # This is not ideal since this error could occur for other reasons.
+        branch = self.branch + ':600'
```

Comment:
The default if no number is given is to already download `lastSuccessfulBuild`, so this test needs to distinguish between downloading a specific build-number, and the default, so something like previous-successful-build-number